src/main/kotlin/com/ampnet/reportservice/service/impl/XlsxServiceImpl.kt

```diff
@@ -0,0 +1,144 @@
+package com.ampnet.reportservice.service.impl
+
+import com.ampnet.reportservice.controller.pojo.XlsxType
+import com.ampnet.reportservice.exception.ErrorCode
+import com.ampnet.reportservice.exception.InternalException
+import com.ampnet.reportservice.grpc.blockchain.BlockchainService
+import com.ampnet.reportservice.grpc.userservice.UserService
+import com.ampnet.reportservice.grpc.wallet.WalletService
+import com.ampnet.reportservice.service.XlsxService
+import com.ampnet.reportservice.service.data.millisecondsToLocalDateTime
+import com.ampnet.userservice.proto.UserExtendedResponse
+import com.ampnet.userservice.proto.UserResponse
+import org.apache.poi.ss.usermodel.CellStyle
+import org.apache.poi.ss.usermodel.Row
+import org.apache.poi.xssf.usermodel.XSSFSheet
+import org.apache.poi.xssf.usermodel.XSSFWorkbook
+import org.springframework.stereotype.Service
+import java.io.ByteArrayOutputStream
+import java.io.IOException
+import java.time.format.DateTimeFormatter
+import java.util.UUID
+import kotlin.jvm.Throws
+
+@Service
+class XlsxServiceImpl(
+    private val userService: UserService,
+    private val walletService: WalletService,
+    private val blockchainService: BlockchainService
+) : XlsxService {
+
+    @Suppress("MagicNumber")
```

Comment:
If you move this to a companion object and make it a `const`, then you may not need `@Suppress("MagicNumber")`